### ISSUE TITLE
[BUG] Error for `gym` with version 0.26.2

### ISSUE THREAD

**Issue by Mamba413** [takuseno/d3rlpy#261]
I find `d3rlpy` is incompatible with `gym` with version 0.26.2.

The code for reproducing the error:
```python
import os
os.chdir(os.path.dirname(os.path.abspath(__file__)))

import gym

# for training
env = gym.make("CartPole-v0")

# for evaluation
eval_env = gym.make("CartPole-v0")

import d3rlpy

# if you don't use GPU, set use_gpu=False instead.
dqn = d3rlpy.algos.DQN(
    batch_size=32,
    learning_rate=2.5e-4,
    target_update_interval=100,
    use_gpu=False,
)

# initialize neural networks with the given environment object.
# this is not necessary when you directly call fit or fit_online method.
dqn.build_with_env(env)

# experience replay buffer
buffer = d3rlpy.online.buffers.ReplayBuffer(maxlen=100000, env=env)

# exploration strategy
# in this tutorial, epsilon-greedy policy with static epsilon=0.3
explorer = d3rlpy.online.explorers.ConstantEpsilonGreedy(0.3)

dqn.fit_online(
    env,
    buffer,
    explorer,
    n_steps=100000,  # train for 100K steps
    eval_env=eval_env,
    n_steps_per_epoch=1000,  # evaluation is performed every 1K steps
    update_start_step=1000,  # parameter update starts after 1K steps
)
```

Error message:
```python
Traceback (most recent call last):
  File "/Users/abc/Documents/MissA-RL/cartpole-d3rlpy.py", line 33, in <module>
    dqn.fit_online(
  File "/Users/abc/.pyenv/versions/3.8.13/envs/missa/lib/python3.8/site-packages/d3rlpy/algos/base.py", line 251, in fit_online
    train_single_env(
  File "/Users/abc/.pyenv/versions/3.8.13/envs/missa/lib/python3.8/site-packages/d3rlpy/online/iterators.py", line 203, in train_single_env
    observation = observation.astype("f4")
AttributeError: 'tuple' object has no attribute 'astype'
```

It is because gym (version 0.26.2) returns two objects and they are stacked into `observation`. It seems that simply change this line:
```
observation = env.reset()
```
into
```python
observation, _ = env.reset()
```
can address this bug.

**Comment by N0Dr4m4L4m4** [comment]
Hey @Mamba413 ,

yea its because of the gym version. The newer version has a different reset return. Like in this issue #238. Try to downgrade gym or just make some changes in the source code.

Hope the best 🎉🤞

**Comment by emrul** [comment]
@N0Dr4m4L4m4 you can try a wrapper to fix this: 
```python
from typing import Any

import gymnasium as gym
from gymnasium.core import WrapperObsType
from gymnasium.wrappers.step_api_compatibility import StepAPICompatibility
class LegacyGymWrapper(StepAPICompatibility):
    def __init__(self, env: gym.Env, output_truncation_bool: bool = False):
        super().__init__(env, output_truncation_bool=output_truncation_bool)

    def reset(self, *, seed: int | None = None, options: dict[str, Any] | None = None) -> tuple[WrapperObsType, dict[str, Any]]:
        obs, _ = super().reset(seed=seed, options=options)
        return obs
```

**Comment by takuseno** [comment]
Feel free to reopen this if you have any further discussion. Please note that the nightly branch has been merged and the latest update `v2.0.2` has been released. In this version, d3rlpy only supports `gym=0.26.0` and drops legacy support.
